examples/client/client/tasks.nim

```diff
@@ -95,6 +100,16 @@ proc statusContext*(arg: ContextArg) {.async, gcsafe, nimcall,
   # re/set threadvars that don't persist across waku dis/connect
   resetContext()
 
+proc updatePrices() {.async.} =
+  updatingPrices = true
+
+  while statusState == StatusState.loggedin:
+    let res = await status.updatePrices()
+    if res.isErr: error "failed to update prices", error = $res.error
```

Comment:
If/when it fails, we should probably send an event to the TUI so the user can be made aware of it.

Reply:
I don't have a strong opinion either way, but `updatePrices` is executed as a background operation, so it may seem odd that the user sees a message for an operation they did not start. However, it could be a message that is displayed in the "status bar" in the future. I'll add it in.

Reply:
Actually, it's not that simple to add in because this is not within the context of a task. Will add a comment and leave it for now to get this merged.